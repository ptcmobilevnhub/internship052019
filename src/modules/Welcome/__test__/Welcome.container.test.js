import React from 'react';
import { shallow, render, mount, configure } from 'enzyme'

import WelcomeContainer from '../Welcome.container'
import { TouchableOpacity } from 'react-native';


import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
global.shallow = shallow;
global.render = render;
global.mount = mount;

describe('Welcome Container Test', () => {
    it('should render snapshot test for welcome container', () => {
        const props = {
            navigation: {
                navigate: jest.fn()
            },
        }
        const wrapper = render(<WelcomeContainer {...props} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should show state as message expect when mount component', () => {
        const props = {
            navigation: {
                navigate: jest.fn()
            }
        }
        const wrapper = mount(<WelcomeContainer {...props} />)
        wrapper.find(TouchableOpacity).props().onPress()

        expect(props.navigation.navigate).toHaveBeenCalledWith('Register')
    })
})