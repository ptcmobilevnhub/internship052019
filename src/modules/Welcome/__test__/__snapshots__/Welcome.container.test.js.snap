// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Container Test should render snapshot test for welcome container 1`] = `
<view
  style="flex:1;margin-horizontal:20px;flex-direction:column;justify-content:space-evenly"
>
  <img
    resizemode="contain"
    source="[object Object]"
    style="height:46px;width:137px"
  />
  <img
    resizemode="contain"
    source="[object Object]"
    style="width:100%"
  />
  <text
    style="font-size:20px;font-weight:bold"
  >
    Your ride, on demand
  </text>
  <text
    style="font-size:10px"
  >
    Whether you're headed to work, the airport, or out on the town, Uber connects you with reliable ride in minutes. One tap and a car comes directly to you.
  </text>
  <view
    style="flex-direction:row;background-color:#000000;border-radius:10px;padding:10px;padding-right:20px;opacity:1"
  >
    <text
      style="flex:1;text-align:center;color:white"
    >
      Register by Phone
    </text>
    <img
      source="[object Object]"
    />
  </view>
</view>
`;
